src/Raven.Server/Documents/ETL/EtlLoader.cs

```diff
@@ -794,6 +818,13 @@ private void LogLongRunningDisposeIfNeeded(Stopwatch sp, string processName)
                 if (existing != null)
                     differences = rabbitMqEtl.Configuration.Compare(existing, transformationDiffs);
             }
+            else if (process is AzureQueueStorageEtl azureQueueStorageEtl)
```

Comment:
This is getting really hard to maintain. Can we have some more generic solution for that? So we won't have it implemented with `else if .. else if ... else if`?

Reply:
That's correct, and there'll be more ETLs soon anyway
We may create a helper class that'll handle these cases or at least extract the common handling to another method.
inb4 I've checked what'll fit here and creating abstract base & custom implementations won't work for `EtlLoader`

Reply:
It will be handled separately